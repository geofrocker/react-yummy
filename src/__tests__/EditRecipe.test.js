import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import EditRecipe from '../components/edit_recipe';
import toJson from 'enzyme-to-json';

describe('edit recipe component', () => {
    const props = {
        history: {
            push: () => {}
        },
        location:{
            search: {}
        },
        match:{
            params:{}
        },
    }
    const wrapper = shallow(<EditRecipe {...props}/>);
    const preventDefault = jest.fn();
    
    it('renders the EditRecipe class', () => {
        expect(wrapper.find(".EditRecipe")).toHaveLength(1);
    });
    it('renders seven div jsx elements', () => {
        expect(wrapper.find("div")).toHaveLength(7);        
    });

    it('renders the edit recipe form and submits data', () =>{
        wrapper.setState({title:'Awesome recipe', category:'General', steps:'one and two', ingredients:"one and two", status:"public",status:[],catData:[]});
        wrapper.find("#recipe-form").simulate('submit', {preventDefault});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(preventDefault).toBeCalled();
    });

    it('render the inputs and hides password', () =>{
        expect(wrapper.find("#title")).toHaveLength(1);
        expect(wrapper.find("#category")).toHaveLength(1);
        expect(wrapper.find("#ingredients")).toHaveLength(1);
        expect(wrapper.find("#steps")).toHaveLength(1);
        expect(wrapper.find("#status")).toHaveLength(1);
    });

    it('renders two select jsx elements', () => {
        expect(wrapper.find("select")).toHaveLength(2);        
    });

    it('renders four options jsx elements', () => {
        expect(wrapper.find("option")).toHaveLength(2);        
    });

    it('renders two buttons jsx elements', () => {
        expect(wrapper.find(".btn")).toHaveLength(2);        
    });
   
})
