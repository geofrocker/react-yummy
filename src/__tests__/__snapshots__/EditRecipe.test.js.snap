// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit recipe component renders the edit recipe form and submits data 1`] = `
<div
  className="EditRecipe"
>
  <h1>
    Edit Recipe
  </h1>
  <div />
  <div
    className="jumbotron col-sm-8"
  >
    <form
      id="recipe-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          required={true}
          type="text"
          value="Awesome recipe"
        />
      </div>
      <div
        className="form-group"
      >
        <select
          className="form-control"
          id="category"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
            value="Select"
          >
            Select Category
          </option>
        </select>
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="ingredients"
          onChange={[Function]}
          required={true}
          type="text"
          value="one and two"
        />
      </div>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="steps"
          onChange={[Function]}
          required={true}
          value="one and two"
        />
      </div>
      <select
        id="status"
        onChange={[Function]}
      >
        <option
          disabled={true}
          selected={true}
          value="Select"
        >
          Select status
        </option>
      </select>
      <br />
      <br />
      <input
        className="btn btn-primary"
        type="submit"
        value="Submit"
      />
      Â 
      <a
        className="btn btn-success"
        href="/dashboard?tab=1"
      >
        Cancel
      </a>
    </form>
  </div>
</div>
`;
